docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+    "\n",
+    "### From Python Sequences or NumPy Arrays\n",
+    "\n",
+    "#### Arrays: N-D Variables\n",
+    "\n",
+    "Variables can be constructed from any Python object that can be used to create a NumPy array or NumPy arrays directly.\n",
+    "See [NumPy array creation](https://numpy.org/doc/stable/user/basics.creation.html) for details.\n",
+    "Given such an object, an array variable can be created using [scipp.array](../../generated/functions/scipp.array.rst) (not to be confused with [data arrays](../../generated/classes/scipp.DataArray.rst)!)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "v1d = sc.array(dims=['x'], values=[1, 2, 3, 4])\n",
+    "v2d = sc.array(dims=['x', 'y'], values=[[1, 2], [3, 4]])\n",
+    "v3d = sc.array(dims=['x', 'y', 'z'], values=[[[1, 2], [3, 4]],\n",
+    "                                             [[5, 6], [7, 8]]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively, passing a NumPy array:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "a = np.array([[1, 2], [3, 4]])\n",
+    "v = sc.array(dims=['x', 'y'], values=a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that *both* the NumPy array and Python lists are copied into the Scipp variable which leads to some additional memory usage.\n",
+    "\n",
+    "The `dtype` and unit of the variable are deduced automatically in the above cases:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "v"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If required, they can be specified directly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.array(dims=['x', 'y'], values=[[1, 2], [3, 4]], dtype='float64', unit='m')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Variances can be added using the `variances` keyword:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "sc.array(dims=['x', 'y'], values=[[1.0, 2.0], [3.0, 4.0]],\n",
+    "         variances=[[0.1, 0.2], [0.3, 0.4]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Scalars: 0-D Variables\n",
+    "\n",
+    "Scalars are variables with no dimensions.\n",
```

Comment:
Should we distinguish more clearly from the nature of the *elements*? We can have a scalar variable containing non-scalar elements (vectors).

Reply:
Not sure how? See also the added listing in `data-structures.ipynb`.

Reply:
Just mention it?